independent-projects/arc/runtime/src/main/java/io/quarkus/arc/ArcUndeclaredThrowableException.java

```diff
@@ -0,0 +1,28 @@
+package io.quarkus.arc;
+
+/**
+ * Exception that is thrown from generated arc classes if a checked exception cannot be propagated
+ */
+public class ArcUndeclaredThrowableException extends RuntimeException {
+
+    public ArcUndeclaredThrowableException(Throwable cause) {
```

Comment:
I'm wondering if the name shouldn't be more general, something like `ArcException` so that we could use it for similar use cases. Note that we have such exception in Weld too: `org.jboss.weld.exceptions.WeldException`.